### ISSUE TITLE
Refunds

### ISSUE THREAD

**Issue by birkjernstrom** [polarsource/polar#4471]
Currently, refunds are rare and we help developers with them quickly (Discord or email – support@polar.sh). However, it's definitely a feature we need to offer to developers directly.

### Must Haves

- [ ] polarsource/polar#4470 to streamline workflow below
- [ ] Trigger refund of an entire order
- [ ] Revoke all associated benefit grants
- [ ] Cancel ongoing subscription effective immediately (if applicable)
- [ ] Send refund confirmation email to customer

### Should Haves

- Developer to provide a reason & comment for the refund
- See above in order history

### Could Haves, e.g could be future iterations

- Partial refund amount
- Setting to disable cancellation of subscription, e.g partial subscription refund
- Setting to disable benefit revokation

**Comment by mrgoonie** [comment]
Yes, definitely needs this feature, I'm working on a product that allows customers to top up credits, sometime they will want to cash out and I don't know how to achieve this with Polar

**Comment by birkjernstrom** [comment]
Will start work on this this week 💪🏼
